File: Guessing_Game.py
#!/usr/bin/python3

from random import randint       # import randint

num_1 = "string"       # Setting value for num_1 & num-2
num_2 = "string"

while True:
    num_1 = input("What is your first number: ")
    if not num_1.isdigit():                        #check if number is vailed
        print()                     #Prints new line
        print("Erorr: Not a vailed number")
        continue
    num_1 = int(num_1)   # converting str to int

    if num_1 > 1000:
        print("Error: number must be less than 1000")
    elif num_1 < -1000:
        print("Error: number must be more than -1000")
    num_2 = input("What is your second number: ")

    if not num_2.isdigit():
        print("Erorr: Not a vailed number")
        continue
    num_2 = int(num_2)   # converting str to int
    
    if num_2 > 1000:
        print("Error: number must be less than 1000")
    elif num_2 < -1000:
        print("Error: number must be more than -1000")
    if num_2 - (num_1 - 1) < 10:
        print("Error: range must be at least 10")
        continue
    else:
        break
        


def p1():
    guess = int(input("What is your guess: "))
    if guess > number:
        print("The number is lower.")          # Player 1 
        print()
    elif guess < number:
        print("The number is higher.")
        print()
    elif guess == "exit":
        exit()
    else:
        print("You Guessed right")
        print("The number was", number)
        exit()

def CPU1():
    global num_1, num_2, cpu_number
    cpu_number = randint(num_1, num_2)
    print("CPU's turn:", cpu_number)       #CPU
    print()
    if cpu_number == number:          
        print("CPU Wins!")
        print("The number was", number)
        exit()

def CPU2():
    global num_1, num_2, cpu_number
    cpu_number = randint(num_1, num_2)
    print("CPU's turn:", cpu_number)
    print()
    if cpu_number > number:
        num_2 = cpu_number - 1                 # CPU 2
    elif cpu_number < number:
        num_1 = cpu_number + 1
    else:
        print("CPU Wins!")
        print("The number was", number)
        exit()


number = randint(num_1, num_2)
guess = "str"        # Setting guess value to a number that will
cpu_number = "str"   # will not be randomized number

print("Number Generating...")          

CPU_level = input("What level would you like to use (1 or 2):")

if CPU_level == str(1):
    while True:
        p1()
        CPU1()
elif CPU_level == str(2):
    while True:
        p1()
        CPU2()

File: GuessingGame.java
import java.util.Scanner;
import java.util.Random;

public class Main {

    public static void CPU1(int tries, int randomNum, Scanner scanner, Random rand, int rangeMin, int rangeMax) {
        int guess;
        while (true) {
            System.out.print("Enter your guess: ");
            if (scanner.hasNextInt()) {
                guess = scanner.nextInt();
            } else {
                System.out.println("Please enter a valid whole number.");
                scanner.next();
                continue;
            }
            tries++;
            if (guess == randomNum) {
                System.out.println("You guessed correctly! The number is " + randomNum);
                System.out.println("Tries: " + tries);
                break;
            } else if (guess < randomNum) {
                System.out.println("Too low, try again.");
            } else {
                System.out.println("Too high, try again.");
            }

            int CPUguess = rand.nextInt(rangeMin, rangeMax + 1);
            if (CPUguess == randomNum) {
                System.out.println("CPU guessed correctly! The number is " + randomNum);
                System.out.println("Tries: " + tries);
                break;
            } else {
                System.out.println("CPU's guess: " + CPUguess);
            }
        }
    }

    public static void CPU2(int tries, int randomNum, Scanner scanner, Random rand, int rangeMin, int rangeMax) {
        int guess;
        while (true) {
            System.out.print("Enter your guess: ");
            if (scanner.hasNextInt()) {
                guess = scanner.nextInt();
            } else {
                System.out.println("Please enter a valid whole number.");
                scanner.next();
                continue;
            }
            tries++;
            if (guess == randomNum) {
                System.out.println("You guessed correctly! The number is " + randomNum);
                System.out.println("Tries: " + tries);
                break;
            } else if (guess < randomNum) {
                System.out.println("Too low, try again.");
            } else {
                System.out.println("Too high, try again.");
            }

            int CPUguess = rand.nextInt(rangeMin, rangeMax + 1);
            if (CPUguess == randomNum) {
                System.out.println("CPU guessed correctly! The number is " + randomNum);
                System.out.println("Tries: " + tries);
                break;
            } else if (CPUguess < randomNum) {
                rangeMin = CPUguess + 1;
            } else {
                rangeMax = CPUguess - 1;
            }
            System.out.println("CPU's guess: " + CPUguess);
        }
    }

    public static void CPU3(int tries, int randomNum, Scanner scanner, Random rand, int rangeMin, int rangeMax) {
        int CPUguess, guess;
        while (true) {
            System.out.print("Enter your guess: ");
            if (scanner.hasNextInt()) {
                guess = scanner.nextInt();
            } else {
                System.out.println("Please enter a valid whole number.");
                scanner.next();
                continue;
            }
            tries++;
            if (guess == randomNum) {
                System.out.println("You guessed correctly! The number is " + randomNum);
                System.out.println("Tries: " + tries);
                break;
            } else if (guess < randomNum) {
                System.out.println("Too low, try again.");
            } else {
                System.out.println("Too high, try again.");
            }

            CPUguess = (rangeMin + rangeMax) / 2;
            if (CPUguess == randomNum) {
                System.out.println("CPU guessed correctly! The number is " + randomNum);
                System.out.println("Tries: " + tries);
                break;
            } else if (CPUguess < randomNum) {
                rangeMin = CPUguess + 1;
            } else {
                rangeMax = CPUguess - 1;
            }
            System.out.println("CPU's guess: " + CPUguess);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Random rand = new Random();
        int rangeMin, rangeMax, tries = 0, CPU;

        while (true) {
            System.out.print("Enter the range min: ");
            if (scanner.hasNextInt()) {
                rangeMin = scanner.nextInt();
            } else {
                System.out.println("Please enter a valid whole number.");
                scanner.next();
                continue;
            }

            System.out.print("Enter the range max: ");
            if (scanner.hasNextInt()) {
                rangeMax = scanner.nextInt();
            } else {
                System.out.println("Please enter a valid whole number.");
                scanner.next();
                continue;
            }

            if (rangeMin > rangeMax) {
                System.out.println("Min must be less than max.");
            } else {
                break;
            }
        }

        System.out.print("Choose a CPU (1-3): ");
        CPU = scanner.nextInt();

        int randomNum = rand.nextInt(rangeMax - rangeMin + 1) + rangeMin;

        if (CPU == 1) {
            CPU1(tries, randomNum, scanner, rand, rangeMin, rangeMax);
        } else if (CPU == 2) {
            CPU2(tries, randomNum, scanner, rand, rangeMin, rangeMax);
        } else if (CPU == 3) {
            CPU3(tries, randomNum, scanner, rand, rangeMin, rangeMax);
        } else {
            System.out.println("Invalid CPU selection. Please choose 1, 2, or 3.");
        }

        scanner.close();
    }
}
